scenarioo-server/src/main/java/org/scenarioo/business/builds/BuildImporter.java

```diff
@@ -120,17 +120,37 @@ public synchronized int submitUnprocessedBuildsForImport(AvailableBuildsList ava
 		List<BuildImportSummary> buildsSortedByDateDescending = BuildByDateSorter
 			.sortBuildsByDateDescending(buildImportSummaries.values());
 
-		List<BuildIdentifier> importNeededBuilds = new LinkedList<BuildIdentifier>();
+		List<BuildIdentifier> importNeededBuilds = new LinkedList<>();
 		for (BuildImportSummary buildImportSummary : buildsSortedByDateDescending) {
 			if (buildImportSummary != null && buildImportSummary.getStatus().isImportNeeded()) {
 				importNeededBuilds.add(buildImportSummary.getIdentifier());
 				submitBuildForImport(availableBuilds, buildImportSummary.getIdentifier());
 			}
 		}
+		submitImportFinishedJob();
 
 		return importNeededBuilds.size();
 	}
 
+	/**
+	 * After all Builds were imported and all Comparisons triggered, we want to log a successful completion message.
+	 * Since the scheduling of comparisons only  happens after all builds are imported, we need to add a second
+	 * Job into the queue to ensure that it is triggered last.
+	 *
+	 * The flow looks like this:
+	 * 1. All Builds are executed and every Build schedules his Comparisons with the AsyncBuildImporterExecutor to ensure
+	 *    that they are only processed after all Builds have been imported.
+	 * 2. The first submit of submitImportFinishedJob is triggered, it schedules another submit, since the Comparisons have not finished yet.
+	 * 3. All Comparisons are scheduled in ComparisonExecutor
+	 * 4. The second submit of submitImportFinishedJob is triggered, it schedules a submit in ComparisonExecutor.
+	 * 5. ComparisonExecutor processes all Comparisons
+	 * 6. ComparisonExecutor processes scheduleComparisonFinishedLog.
+	 */
```

Comment:
I wouldnt expect a such a long comment on a private method to explain all the outer context and the full flow. I think this can be drastically shortened and/or documented in another place, because this private method only does a small part of all of this.